packages/devtools-reps/src/object-inspector/utils/node.js

```diff
@@ -619,101 +608,253 @@ function setNodeChildren(
 }
 
 function getChildren(options: {
-  actors: Object,
-  getObjectEntries: (RdpGrip) => ?LoadedEntries,
-  getObjectProperties: (RdpGrip) => ?LoadedProperties,
+  cachedNodes: CachedNodes,
+  loadedProperties: LoadedProperties,
   item: Node
 }) : Array<Node> {
   const {
-    actors = {},
-    getObjectEntries,
-    getObjectProperties,
-    item
+    cachedNodes,
+    loadedProperties = new Map(),
+    item,
   } = options;
+
+  const key = item.path;
+  if (cachedNodes && cachedNodes.has(key)) {
+    return cachedNodes.get(key);
+  }
+
+  const loadedProps = loadedProperties.get(key);
+  const {
+    ownProperties,
+    ownSymbols,
+    safeGetterValues,
+    prototype
+  } = loadedProps || {};
+  const hasLoadedProps = ownProperties || ownSymbols || safeGetterValues || prototype;
+
+  // Because we are dynamically creating the tree as the user
+  // expands it (not precalculated tree structure), we cache child
+  // arrays. This not only helps performance, but is necessary
+  // because the expanded state depends on instances of nodes
+  // being the same across renders. If we didn't do this, each
+  // node would be a new instance every render.
+  // If the node needs properties, we only add children to
+  // the cache if the properties are loaded.
+  const addToCache = (children: Array<Node>) => {
```

Comment:
I think we'll unmount the component when we resume, and if the cache is in the component we're safe...